booking-flight-service/src/test/java/com/smoothstack/booking/BookingIntegrationTests.java

```diff
@@ -0,0 +1,353 @@
+        .content(mapper.writeValueAsString(booking))
+        .contentType(MediaType.APPLICATION_JSON)
+      )
+        .andExpect(status().isNotFound());
+  }
+
+  @DisplayName("test update booking nost authorized user, expect forbidden")
+  @Test
+  @Order(4)
+  void testUpdateBookingForbidden() throws Exception {
+    Booking booking = new Booking(BOOKING_ID.longValue(), false, "payment", LOGGED_IN_USER_ID.intValue(), List.of(), Set.of());
+    // BookingRequest bookingRequest = new BookingRequest(booking, List.of(TRAVELER_ID));
+    mockMvc.perform(
+      put("/booking")
+        .header("user-id", LOGGED_IN_USER_ID_NOT_AUTHORIZED)
+        .header("user-role", "USER")
+        .content(mapper.writeValueAsString(booking))
+        .contentType(MediaType.APPLICATION_JSON)
+      )
+        .andExpect(status().isForbidden());
+  }
+
+  @DisplayName("test update booking as not user owner with employee role, expect 200")
+  @Test
+  @Order(4)
+  void testUpdateBookingEmployeeRole() throws Exception {
+    Booking booking = new Booking(BOOKING_ID.longValue(), false, "payment", LOGGED_IN_USER_ID.intValue(), List.of(), Set.of());
+    // BookingRequest bookingRequest = new BookingRequest(booking, List.of(TRAVELER_ID));
+    mockMvc.perform(
+      put("/booking")
+        .header("user-id", LOGGED_IN_USER_ID_NOT_AUTHORIZED)
+        .header("user-role", "EMPLOYEE")
+        .content(mapper.writeValueAsString(booking))
+        .contentType(MediaType.APPLICATION_JSON)
+      )
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.isActive", is(false)));
+  }
+
+  @DisplayName("test delete booking as a valid user, expect OK   ")
+  @Test
+  @Order(6)
+  void testDeleteBooking() throws Exception {
+    mockMvc.perform(
+      delete("/booking/" + "1")
+        .header("user-id", LOGGED_IN_USER_ID)
+        .header("user-role", "USER")
+      )
+        .andExpect(status().isOk());
+  }
+
+  @DisplayName("test delete booking that does not exist, expect not found")
+  @Test
+  @Order(7)
+  void testDeleteBookingNotFound() throws Exception {
+    mockMvc.perform(
+      delete("/booking/" + USER_ID_NO_BOOKINGS)
+        .header("user-id", LOGGED_IN_USER_ID)
+        .header("user-role", "USER")
+      )
+        .andExpect(status().isNotFound());
+  }
+
+  @DisplayName("test delete booking not authorized, expect forbidden")
+  @Test
+  @Order(5)
+  void testDeleteBookingNotAuthorized() throws Exception {
+    mockMvc.perform(
+      delete("/booking/" + BOOKING_NOT_OWNED_ID)
+        .header("user-id", LOGGED_IN_USER_ID_NOT_AUTHORIZED)
+        .header("user-role", "USER")
+      )
+        .andExpect(status().isForbidden());
+  }
+
+  @DisplayName("test delete booking with user role EMPLOYEE, expect 200")
+  @Test
+  @Order(6)
+  void testDeleteBookingWithEmployeeRole() throws Exception {
+    mockMvc.perform(
+      delete("/booking/" + BOOKING_NOT_OWNED_ID)
+        .header("user-id", LOGGED_IN_USER_ID_NOT_AUTHORIZED)
```

Comment:
should use a better user id for this test, the name of this variable makes me think the test should return unauthorized